Feature/netcore1.0

alyiox opened the pull request:
1. Migration from DNX RC1 to NET Core CLI.
2. Target framework to `netstandard1.0`
3. Support to custom cache policy by `CacheEntryOptions`
4. Update version from `1.2.0-beta1` to `1.2.1`

alyiox committed:
Migration from DNX RC1 to NET Core 1.0

alyiox committed:
Fix MemoryCacheBasedQueryCache.cs

- Fix unit tests.

alyiox committed:
- Support to custom cache policy on each cache entry with CacheEntryOptions.

alyiox committed:
Add test case .

karldodd commented on src/DotQuery.Extensions/MemoryCacheBasedQueryCache.cs:
```diff
@@ -1,43 +1,56 @@
         }
 
-        public TValue GetOrAdd(TKey key, TValue lazyTask)
+        public TValue GetOrAdd(TKey key, TValue lazyTask) => GetOrAdd(key, lazyTask, new CacheEntryOptions { SlidingExpiration = _slidingExpiration });
```
Avoid using 'new CacheEntryOptions' if we can reuse an instance

alyiox replied:
fixed

karldodd commented on src/DotQuery.Extensions/project.json:
```diff
@@ -1,32 +1,29 @@
 {
-  "version": "1.2.0-beta1",
+  "version": "1.2.1",
```
Let's make it to 1.3 as upgrading to .net core 1.0 is a huge step forward.

alyiox replied:
fixed

karldodd commented on src/DotQuery.Core/AsyncQueryExecutor.cs:
```diff
@@ -42,6 +42,23 @@ public Task<TResult> QueryAsync(TQuery query)
+        /// <param name="query">The query to be executed</param>
+        /// <param name="options">The cache policy options.</param>
+        /// <returns>The task object representing the asynchronous operation</returns>
+        public Task<TResult> QueryAsync(TQuery query, CacheEntryOptions options)
```
Any reason we need a standalone `CacheEntryOptions`? Would be great if we integrate cache entry option into exsiting QueryOptions class (as a field like CacheOption). @qqbuby

alyiox replied:
It's a pity that `QueryOptions` is a flags `enum` type, not a `class`. So create a new method signature with `QueryAsync(TQuery, CacheEntryOptions)` for the  the backward compatibility.

karldodd replied:
@qqbuby Try merge QueryOptions and CacheEntryOptions to a single unified class

alyiox committed:
Update versiont form 1.2.1 to 1.3.0

- Remove 'QueryAsync(TQuery query, CacheEntryOptions options)' from 'IAsyncQueryExecutor<TQuery, TResult>'

alyiox committed:
Redesign codebase.

alyiox committed:
Refactor codebase

- Add more test cases

karldodd commented:
Will merge and release an beta version @qqbuby
